packages/compass-components/src/components/document-list/document-actions-group.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { css } from '@leafygreen-ui/emotion';
+import { spacing } from '@leafygreen-ui/tokens';
+import { Button, Icon } from '../leafygreen';
+
+const actionsGroupContainer = css({
+  position: 'absolute',
+  display: 'flex',
+  gap: spacing[2],
+  width: '100%',
+  top: spacing[3],
+  paddingLeft: spacing[3],
+  paddingRight: spacing[3],
+  pointerEvents: 'none',
+});
+
+const actionsGroupItem = css({
+  flex: 'none',
+  pointerEvents: 'all',
+});
+
+const actionsGroupItemSeparator = css({
+  flex: '1 0 auto',
+});
+
+function useElementParentHoverState<T extends HTMLElement>(
+  ref: React.RefObject<T>
+): boolean {
+  const [isHovered, setIsHovered] = useState(false);
+
+  useEffect(() => {
+    const node = ref.current?.parentElement;
+
+    const onMouseEnter = () => {
+      setIsHovered(true);
+    };
+
+    const onMouseLeave = () => {
+      setIsHovered(false);
+    };
+
+    node?.addEventListener('mouseenter', onMouseEnter);
+    node?.addEventListener('mouseleave', onMouseLeave);
+
+    return () => {
+      node?.removeEventListener('mouseenter', onMouseEnter);
+      node?.removeEventListener('mouseleave', onMouseLeave);
+    };
+  }, [ref]);
+
+  return isHovered;
+}
+
+const DocumentActionsGroup: React.FunctionComponent<
+  {
+    onEdit?: () => void;
+    onCopy?: () => void;
+    onClone?: () => void;
+    onRemove?: () => void;
+    showOnHover?: boolean;
+  } & (
+    | { onExpand?: never; expanded?: never }
+    | { onExpand: () => void; expanded: boolean }
+  )
+> = ({
+  onEdit,
+  onCopy,
+  onClone,
+  onRemove,
+  onExpand,
+  expanded,
+  showOnHover = true,
+}) => {
+  const conatinerRef = useRef<HTMLDivElement | null>(null);
+  const isHovered = useElementParentHoverState(conatinerRef);
+
+  return (
+    <div
+      ref={conatinerRef}
+      className={actionsGroupContainer}
+      style={{ display: showOnHover ? (isHovered ? 'flex' : 'none') : 'flex' }}
+    >
+      {onExpand && (
+        <Button
+          size="xsmall"
+          rightGlyph={
+            <Icon glyph={expanded ? 'CaretDown' : 'CaretRight'}></Icon>
+          }
+          title={expanded ? 'Collapse all' : 'Expand all'}
```

Comment:
Should we add an `aria-label` in addition to the `title` for these items?
Looks like it might not hurt?
https://dev.opera.com/articles/ux-accessibility-aria-label/#accessible-name-calculation
https://www.w3.org/WAI/tutorials/forms/labels/

Reply:
Titles are read by voice over, I did forget to disable icon title which leafygreen doesn't do by default (and then they are weirdly read by VO), will fix